### ISSUE TITLE
Have `Residual` post-processor work with ActuallyExplicitEuler

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#11857]
## Rationale

This would have the post-processor `Residual` (which gets the value of the current nonlinear residual) work with `ActuallyExplicitEuler`. This is very useful for applications with nonlinear equation sets that potentially use this time integrator, as it can verify that the system being solved is actually linear by verifying that the nonlinear residual at the end of the step is sufficiently close to zero.

## Description

Since `ActuallyExplicitEuler` does not monitor the nonlinear residual at the end of the time step, there would need to be some kind of trigger (maybe a new parameter in `ActuallyExplicitEuler` called "monitor_nonlinear_residual") that when true would cause the nonlinear residual to be evaluated after the single linear solve.

Then it would be *nice* to be able to have `Residual` check that the time integrator/`solve_type` is actually monitoring the residual, so that users of `Residual` do not use it incorrectly; however, this is not *absolutely* necessary.

Note that when using some other time integrator and `solve_type=LINEAR`, with the `-snes_monitor_cancel` PETSc flag now being set, the user must cancel the cancellation by adding `-snes_monitor_cancel 0` (which overrides the earlier flag without the "0"). Again, it would be *nice* to have a MOOSE interface for doing this, but it's not necessary.

## Impact

Extend scope of `Residual` post-processor.

**Comment by joshuahansel** [comment]
@andrsd @friedmud This feature is what I was referring to in our discussion yesterday. Does this seem reasonable? I think it'd definitely be useful.

**Comment by friedmud** [comment]
I'm not sure this will work quite right with ActuallyExplicitEuler.  The
reason is that we don't set any of the "implicit = false" flags... so if
you evaluate the nonlinear residual after the solve it will actually
evaluate the _implicit_ nonlinear residual instead of the explicit one.  I
can't see how that would be useful...

**Comment by joshuahansel** [comment]
Yeah, you're right - we'd have to set those flags for any non-time residual objects... so I guess for testing that a system is linear, users should just use `Residual` with `NotActuallyExplicitEuler` (with "implicit = false" as needed), and just infer that the residual should be the same as with `ActuallyExplicitEuler`. Would there be any potential for residual difference between the `Actually` and `NotActually`, such as the handling of constraints? If so, then I'd still suggest the ability to compute the residual after the solve; it would just require users to set those "implicit" flags.

**Comment by friedmud** [comment]
Thinking about this more - it wouldn't be bad actually to compute the
implicit residual... like you say, it could show you how far off you are
from your problem actually being linear.

Can you code this up and try it?
